peps/pep-0733.rst

```diff
@@ -0,0 +1,629 @@
+PEP: 733
+Title: An Evaluation of Python's Public C API
+Author: Erlend E. Aasland <erlend@python.org>, Domenico Andreoli <domenico.andreoli@linux.com>, Stefan Behnel <stefan_ml@behnel.de>, Carl Friedrich Bolz-Tereick <cfbolz@gmx.de>, Simon Cross <hodgestar@gmail.com>, Steve Dower <steve.dower@python.org>, Tim Felgentreff <tim.felgentreff@oracle.com>, David Hewitt <1939362+davidhewitt@users.noreply.github.com>, Shantanu Jain <hauntsaninja at gmail.com>, Wenzel Jakob <wenzel.jakob@epfl.ch>, Irit Katriel <irit@python.org>, Marc-Andre Lemburg <mal@lemburg.com>, Donghee Na <donghee.na@python.org>, Karl Nelson <nelson85@llnl.gov>, Ronald Oussoren <ronaldoussoren@mac.com>, Antoine Pitrou <solipsis@pitrou.net>, Guido van Rossum <guido@python.org>, Neil Schemenauer <nas@arctrix.com>, Mark Shannon <mark@hotpy.org>, Stepan Sindelar <stepan.sindelar@oracle.com>, Gregory P. Smith <greg@krypto.org>, Eric Snow <ericsnowcurrently@gmail.com>, Victor Stinner <vstinner@python.org>, Serhiy Storchaka <storchaka@gmail.com>, Petr Viktorin <encukou@gmail.com>, Carol Willing <willingc@gmail.com>, William Woodruff <william@yossarian.net>, David Woods <dw-git@d-woods.co.uk>, Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 16-Oct-2023
+
+
+Abstract
+========
+
+This **informational** PEP describes our shared view of the public C API. The
+document defines:
+
+* purposes of the C API
+* stakeholders and their particular use cases and requirements
+* strengths of the C API
+* problems of the C API categorized into nine areas of weakness
+
+This document does not propose solutions to any of the identified problems. By
+creating a shared list of C API issues, this document will help to guide
+continuing discussion about change proposals and to identify evaluation
+criteria.
+
+
+Introduction
+============
+
+Python's C API was not designed for the purposes it currently fulfils.
```

Comment:
```suggestion
Python's C API was not designed for the purposes it currently fulfills.
```
The US spelling is usually 'fulfill' rather than 'fulfil' (mostly UK).

Reply:
As Irit is the primary author of the text, I think we should keep the British spellings. I think there's also a "behaviour" further down.

Reply:
I'm French and learnt UK spelling at school. It's very confusing for me when people report me "typos" in my spelling and suggest the US spelling, whereas I learnt correctly the UK spelling :-) I understood that in Python, most people use the US spelling. In general, I mostly care about consistency. For this document, I'm fine with UK or US, as long as it's consistent in the whole document (as @JelleZijlstra says).

Note: I didn't know that fulfils has different spellings!

Reply:
My mother tongue is Hebrew, I learned English in the US and now I like in the UK. So expect my English to be a messy mixture of the two.